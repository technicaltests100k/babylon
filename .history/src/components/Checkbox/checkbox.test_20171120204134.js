import React, { Component } from 'react';

class Checkbox extends Component {
  render() {
    return <div />;
  }
}

export default Checkbox;
