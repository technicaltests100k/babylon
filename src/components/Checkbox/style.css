.checkbox {
  width: 20px;
  height: 20px;
  border-style: inset;
  display: inline-block;
  vertical-align: bottom;
}

.checkbox:hover {
  cursor: pointer;
}

.checkboxIcon {
  width: 20px;
  height: 20px;
  fill: green;
}
