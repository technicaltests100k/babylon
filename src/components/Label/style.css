.label {
  display: block;
  font-size: 22px;
  margin: 20px 0;
}
