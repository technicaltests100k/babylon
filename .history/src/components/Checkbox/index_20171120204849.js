import React, { Component } from 'react';
import style from './style.css';

class Checkbox extends Component {
  render() {
    return <div className={style} />;
  }
}

export default Checkbox;
