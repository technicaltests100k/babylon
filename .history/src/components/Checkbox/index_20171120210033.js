import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor() {
    super(props);
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    return <div className={'checkbox'} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
