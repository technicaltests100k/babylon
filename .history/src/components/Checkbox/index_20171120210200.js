import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor() {
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    return <div className={'checkbox'} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
