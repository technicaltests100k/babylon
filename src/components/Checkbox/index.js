import React, { Component } from 'react';
import './style.css';
import icon from '../../icons.svg';

class Checkbox extends Component {
  constructor() {
    super();
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    const style = this.state.isChecked ? 'checked' : '';
    return (
      <div className={`checkbox ${style}`} onClick={this.handleOnClick}>
        <svg className={'icon'}>
          <use xlinkHref={`${icon}#icon-checkmark`} />
        </svg>
      </div>
    );
  }
}

export default Checkbox;
