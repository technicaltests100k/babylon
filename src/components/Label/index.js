import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.css';

/**
  *	Label React Type component.
  */
export default class Label extends Component {
  /**
  *	Props implementation.
  */
  static propTypes = {
    extendStyle: PropTypes.string,
    htmlFor: PropTypes.string,
    form: PropTypes.string,
    content: PropTypes.string.isRequired,
    children: PropTypes.element
  };

  /**
  * Render
  * @return {ReactElement} markup
  */
  render() {
    const { htmlFor, form, extendStyle, content, children } = this.props;
    return (
      <label className={`label ${extendStyle}`} htmlFor={htmlFor} form={form}>
        {content}
        {children}
      </label>
    );
  }
}
