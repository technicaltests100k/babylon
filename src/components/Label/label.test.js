import React from 'react';
import Label from './';
import Checkbox from '../Checkbox';
/**
* Test for <Label /> component.
* @test {Label}
*/
const props = {
  extendStyle: 'classToExtend',
  htmlFor: 'form1',
  form: 'form2',
  content: 'a super interesting text'
};

describe('<Label />', () => {
  it('should render Label with all the correct props', () => {
    const wrapper = render(<Label {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Label with the checkbox component', () => {
    const wrapper = render(
      <Label {...props}>
        <Checkbox />
      </Label>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
