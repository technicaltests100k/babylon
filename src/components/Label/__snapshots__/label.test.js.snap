// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render Label with all the correct props 1`] = `
<label
  class="label classToExtend"
  for="form1"
  form="form2"
>
  a super interesting text
</label>
`;

exports[`<Label /> should render Label with the checkbox component 1`] = `
<label
  class="label classToExtend"
  for="form1"
  form="form2"
>
  a super interesting text
  <div
    class="checkbox "
  >
    
  </div>
</label>
`;
